fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/RdfStreamContext.java

```diff
@@ -0,0 +1,183 @@
+    }
+
+    /**
+     * Returns the proffered {@link Triple}s with the context of this RdfStreamContext.
+     *
+     * @param stream the stream
+     * @return proffered Triples with the context of this RDFStream
+     */
+    public RdfStreamContext withThisContext(final Stream<Triple> stream) {
+        return new RdfStreamContext(stream).namespaces(namespaces()).topic(topic());
+    }
+
+    /**
+     * Set a {@link Node} context for this stream
+     * @param node the Node
+     * @return the original RdfStreamContext
+     */
+    public RdfStreamContext topic(final Node node) {
+        this.topic = node;
+        return this;
+    }
+
+    /**
+     * Retrieve the {@link Node} context for this stream
+     * @return the topic Node
+     */
+    public Node topic() {
+        return topic;
+    }
+
+    /**
+     * Set a Map of namespace for this stream
+     * @param namespaces the namespace map
+     * @return the original RdfStreamContext
+     */
+    public RdfStreamContext namespaces(final Map<String, String> namespaces) {
+        this.namespaces.putAll(namespaces);
+        return this;
+    }
+
+    /**
+     * Set a single namespace for this stream
+     * @param prefix the namespace prefix
+     * @param uri the full URI for this namespace
+     * @return the original RdfStreamContext
+     */
+    public RdfStreamContext namespace(final String prefix, final String uri) {
+        this.namespaces.put(prefix, uri);
+        return this;
+    }
+
+    /**
+     * @return Namespaces in scope for this stream.
+     */
+    public Map<String, String> namespaces() {
+        return namespaces;
+    }
+
+    /**
+     * Set the session context for this stream
+     * @param session the Jcr Session
+     * @return the original RdfStreamContext
+     */
+    public RdfStreamContext context(final Session session) {
+        this.context = session;
+        return this;
+    }
+
+    /**
+     * Retrieve the {@link Session} context for this stream
+     * @return the Jcr Session
+     */
+    public Session context() {
+        return context;
+    }
+
+    /**
+     * A namespace-aware {@link Collector}
+     * @return a Collector for use with aggregating the Triple stream.
+     */
+    public Collector<Triple, ?, Model> toModel() {
+        final Model model = createDefaultModel();
+        model.setNsPrefixes(namespaces());
+        return Collector.of(() -> model,
+            (m, t) -> m.add(m.asStatement(t)),
+            (left, right) -> { left.add(right); return left; },
+            Collector.Characteristics.UNORDERED);
+    }
+
+    /**
+     * Create an RdfStreamContext from an existing Model.
+     *
+     * @param model the model
+     * @return the corresponding RdfStreamContext
+     */
+    public static RdfStreamContext fromModel(final Model model) {
+        return new RdfStreamContext(StreamSupport.stream(spliteratorUnknownSize(model.listStatements(), NONNULL), false)
```

Comment:
Nope: https://jira.duraspace.org/browse/FCREPO-1672

I'm working on that, but I'm stuck on how string literals get mapped to/from the JCR layer w/r/t their Datatype. It must be a simple thing that I'm overlooking, but if you'd like to take over that ticket, you'd be welcome to.